### ISSUE TITLE
Inspect the feasibility of using Psalm to decrease run-time errors

### ISSUE THREAD

**Issue by jonasraoni** [pkp/pkp-lib#7591]
**Describe the problem you would like to solve**
Due to the nature of PHP, it's challenging to detect simple run-time errors due to broken typing promises.

**Describe the solution you'd like**
Investigate whether [Psalm](https://psalm.dev/), is able to help us mitigate these errors at building-time.
Perhaps it will produce too much noise due to untyped 3rd-party code.

**Who is asking for this feature?**
Developer

**Comment by NateWr** [comment]
Were you thinking of trying to use it during the validation tests with Travis?

**Comment by jonasraoni** [comment]
Yeah, given that we don't have a tons of unit tests, anything that helps mitigating unexpected errors automatically, sounds great for me.
But just like the issue about "removing deprecated code", this is mainly a self-reminder. Let me know if we have a better place this kind of thing :)

**Comment by NateWr** [comment]
I've have a bunch of these note-to-self issues scattered around too. As long as it's something you really plan to get to, I'm ok leaving them filed here.

**Comment by jonasraoni** [comment]
Ok! So I'll use this issue to keep related resources. In order to auto-refactor code to use newer PHP features (it might probably cover deprecations), we can use the https://github.com/rectorphp/rector
